### PR TITLE
compile/test speed improvements

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: changelog.d/+compile-test-improvements.internal.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+compile/test speed improvements
+
+1. add CARGO_NET_GIT_FETCH_WITH_CLI=true to agent's Dockerfile since we found out it
+    saves a lot of time on fetching (around takes 60s when using libgit2)
```
Comment (meowjesty) [inline changelog.d/+compile-test-improvements.internal.md:4]:
```suggestion
    saves a lot of time on fetching (takes around 60s when using libgit2)
```

**Review by meowjesty — APPROVED** [review]
Typo.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -130,14 +123,9 @@ jobs:
     if: ${{needs.changed_files.outputs.rs_changed == 'true'}}
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
```
Comment (t4lz) [inline .github/workflows/ci.yaml:127]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -150,34 +138,19 @@ jobs:
...
```
Comment (t4lz) [inline .github/workflows/ci.yaml:142]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -150,34 +138,19 @@ jobs:
...
-          profile: minimal
-          toolchain: nightly
           target: ${{matrix.target}}
```
Comment (t4lz) [inline .github/workflows/ci.yaml:143]:
```suggestion
```

**Review by t4lz — COMMENTED** [review]
Some cosmetic suggestions while I continue reviewing.

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -65,12 +51,14 @@ jobs:
...
+        with:
+          target: x86_64-apple-darwin,aarch64-apple-darwin
+          toolchain: nightly-2023-01-31
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
-          profile: minimal
-          toolchain: nightly
-          target: aarch64-apple-darwin
-      - uses: Swatinem/rust-cache@v2
+          target: aarch64-apple-darwin,aarch64-apple-darwin
```
Comment (t4lz) [inline .github/workflows/release.yaml:61]: This looks a bit like a duplicate, is this on purpose?

**Review by t4lz — APPROVED** [review]
Awesome change. Can be merged when the other comment is resolved.

File: .github/workflows/release.yaml (hunk 5 above)
Comment (aviramha) [reply .github/workflows/release.yaml:61]: nice catch!

**Pull request merged**